types/ramda/index.d.ts

```diff
@@ -1309,10 +1309,12 @@ declare namespace R {
 
         /**
          * Returns a function that when supplied an object returns the indicated property of that object, if it exists.
-         * Note: TS1.9 # replace any by dictionary
          */
-        prop<T>(p: string, obj: any): T;
-        prop<T>(p: string): <T>(obj: any) => T;
+        prop<T>(p: string | number, obj: any): T;
+        prop<T>(p: string | number): <T>(obj: any) => T;
```

Comment:
you should put the 2 above at the bottom. It will return better results (as typescript first tries to apply stuff from the top).

how it works now:
```typescript
const obj = {a: 1, b: 2};

// type: {}
const res = r.prop('a', obj);
```

how it works if you put the 2 at the bottom:
```typescript
const obj = {a: 1, b: 2};

// type: number
const res = r.prop('a', obj);
```